dev/stacks/dpe-sandbox-k8s-deployments/main.tf

```diff
@@ -0,0 +1,327 @@
+      metadata {
+        labels = {
+          role = "client"
+        }
+      }
+
+      spec {
+        container {
+          name              = "client"
+          image             = "calico/star-probe:v0.1.0"
+          image_pull_policy = "Always"
+
+          command = ["probe", "--urls=http://frontend.stars:80/status,http://backend.stars:6379/status"]
+
+          port {
+            container_port = 9000
+          }
+        }
+      }
+    }
+  }
+
+}
+
+resource "kubernetes_service" "client-service" {
+  depends_on = [kubernetes_namespace.client]
+  metadata {
+    name      = "client"
+    namespace = "client"
+  }
+
+  spec {
+    selector = {
+      role = "client"
+    }
+
+    port {
+      port        = 9000
+      target_port = 9000
+    }
+  }
+}
+
+resource "kubernetes_service" "frontend-service" {
+  depends_on = [kubernetes_namespace.stars-namespace]
+  metadata {
+    name      = "frontend"
+    namespace = "stars"
+  }
+
+  spec {
+    selector = {
+      role = "frontend"
+    }
+
+    port {
+      port        = 80
+      target_port = 80
+    }
+  }
+}
+
+resource "kubernetes_deployment" "frontend-deployment" {
+  depends_on = [kubernetes_namespace.stars-namespace]
+  metadata {
+    name      = "frontend"
+    namespace = "stars"
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        role = "frontend"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          role = "frontend"
+        }
+      }
+
+      spec {
+        container {
+          name              = "frontend"
+          image             = "calico/star-probe:v0.1.0"
+          image_pull_policy = "Always"
+
+          command = ["probe", "--urls=http://frontend.stars:80/status,http://backend.stars:6379/status,http://client.client:9000/status"]
+
+          port {
+            container_port = 80
+          }
+        }
+      }
+    }
+  }
+}
+
+
+resource "kubernetes_service" "backend-service" {
+  depends_on = [kubernetes_namespace.stars-namespace]
+  metadata {
+    name      = "backend"
+    namespace = "stars"
+  }
+
+  spec {
+    selector = {
+      role = "backend"
+    }
+
+    port {
+      port        = 6379
+      target_port = 6379
+    }
+  }
+}
+
+resource "kubernetes_deployment" "backend-deployment" {
+  depends_on = [kubernetes_namespace.stars-namespace]
+  metadata {
+    name      = "backend"
+    namespace = "stars"
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        role = "backend"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          role = "backend"
+        }
+      }
+
+      spec {
+        container {
+          name              = "backend"
+          image             = "calico/star-probe:v0.1.0"
+          image_pull_policy = "Always"
+
+          command = ["probe", "--http-port=6379", "--urls=http://frontend.stars:80/status,http://backend.stars:6379/status,http://client.client:9000/status"]

```

Comment:
Nice, I'll have to read into calico more but looks like a good open source tool for networking

Reply:
In this case I am only using these k8 resources to easily prove out some networking within the cluster. At this point in time I am not planning to implement custom networking; instead am going to continue using the AWS VPC-CNI plugin for EKS. There is more tweaking I'll need to make to get all of it to work like we want.

There are 2 ways AWS suggests to handle networking:

1. There is pod level security groups - Meaning each pod can be assigned an AWS security group, and we get very fine grained control over it. So we can easily limit pod -> pod connections, pod -> AWS resources, pod -> Internet, Internet -> pod connections
2. They also allow us to use a k8 resource `NetworkPolicy`, but I need to spend more time to understand how it works with security groups & if both as needed; What the best practices are.